lib/tool_shed/galaxy_install/tool_dependencies/recipe/step_handler.py

```diff
@@ -872,6 +873,52 @@ def prepare_step( self, tool_dependency, action_elem, action_dict, install_envir
         return action_dict
 
 
+class RegexReplace( RecipeStep ):
+
+    def __init__( self, app ):
+        self.app = app
+        self.type = 'regex_replace'
+
+    def execute_step( self, tool_dependency, package_name, actions, action_dict, filtered_actions, env_file_builder,
+                      install_environment, work_dir, current_dir=None, initial_download=False ):
+        """
+        Search and replace text in a file using regular expressions. Since this class is not used in the initial
+        download stage, no recipe step filtering is performed here, and None values are always returned for
+        filtered_actions and dir.
+        """
+        log_file = os.path.join( install_environment.install_dir, basic_util.INSTALLATION_LOG )
+        if os.path.exists( log_file ):
+            logfile = open( log_file, 'ab' )
+        else:
+            logfile = open( log_file, 'wb' )
+        if os.path.isabs( action_dict[ 'filename' ] ):
+            filename = action_dict[ 'filename' ]
+            if not ( filename.startswith( current_dir ) or filename.startswith( install_environment.install_dir ) ):
+                return tool_dependency, None, None
+        else:
+            filename = os.path.abspath( os.path.join( current_dir, action_dict[ 'filename' ] ) )
+        regex = re.compile( action_dict[ 'regex' ], flags=re.MULTILINE )
+        replacement = action_dict[ 'replacement' ]
+        haystack = file( filename, 'r' ).read()
```

Comment:
works for me :)

tir. 14. jul. 2015 kl. 12.59 skrev Dave B. <notifications@github.com>:

> In lib/tool_shed/galaxy_install/tool_dependencies/recipe/step_handler.py
> <https://github.com/galaxyproject/galaxy/pull/457#discussion_r34598497>:
>
> > +        filtered_actions and dir.
> > +        """
> > +        log_file = os.path.join( install_environment.install_dir, basic_util.INSTALLATION_LOG )
> > +        if os.path.exists( log_file ):
> > +            logfile = open( log_file, 'ab' )
> > +        else:
> > +            logfile = open( log_file, 'wb' )
> > +        if os.path.isabs( action_dict[ 'filename' ] ):
> > +            filename = action_dict[ 'filename' ]
> > +            if not ( filename.startswith( current_dir ) or filename.startswith( install_environment.install_dir ) ):
> > +                return tool_dependency, None, None
> > +        else:
> > +            filename = os.path.abspath( os.path.join( current_dir, action_dict[ 'filename' ] ) )
> > +        regex = re.compile( action_dict[ 'regex' ], flags=re.MULTILINE )
> > +        replacement = action_dict[ 'replacement' ]
> > +        haystack = file( filename, 'r' ).read()
>
> @erasche <https://github.com/erasche> Point taken. That said, my general
> preference is to err on the side of being more careful than necessary.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/galaxyproject/galaxy/pull/457/files#r34598497>.
>